Gateway/Http/Client/TransactionPayment.php

```diff
@@ -67,12 +85,46 @@ public function placeRequest(TransferInterface $transferObject): mixed
             $client = $this->adyenHelper->initializeAdyenClient();
         }
 
+        $service = $this->adyenHelper->createAdyenCheckoutService($client);
+
+        if (isset($request['giftcardRequestParameters'])) {
+            $redeemedGiftcards = $request['giftcardRequestParameters'];
+            unset($request['giftcardRequestParameters']);
+
+            try {
+                $ordersResponse = $this->orderApiHelper->createOrder(
+                    $request['reference'],
+                    $request['amount']['value'],
+                    $request['amount']['currency'],
+                    $this->storeManager->getStore()->getId()
+                );
+
+                $response = $this->handleGiftcardPayments($request, $service, $redeemedGiftcards, $ordersResponse);
+
+                $request['amount']['value'] = $this->remainingOrderAmount;
+                $request['order'] = [
+                    'pspReference' => $ordersResponse['pspReference'],
+                    'orderData' => $ordersResponse['orderData']
+                ];
+            } catch (AdyenException $e) {
+                $response['error'] = $e->getMessage();
+                $response['errorCode'] = $e->getAdyenErrorCode();
+
+                $this->adyenHelper->logResponse($response);
+
+                return $response;
+            }
+
+            if ($this->remainingOrderAmount === 0) {
+                return $response;
+            }
+        }
```

Comment:
Let's wrap this logic into a function to make the original function more readable

Reply:
Let's discuss it offline.